src/Spectre.libStatistics/Math.h

```diff
@@ -380,4 +380,30 @@ std::vector<DataType> build(size_t size, Generator generator)
     return result;
 }
 
+/// <summary>
+/// Calculates differences between adjacent elements of given container.
+/// </summary>
+/// <param name="data">The container.</param>
+/// <param name="order">Order of difference operation (how many times the difference is executed).</param>
+/// <returns>Vector of calculated differences.</returns>
+ template <class DataType>
+ std::vector<DataType> diff(gsl::span<const DataType> data, uint16_t order = 1)
+ {
+    size_t length = data.length();
+    std::vector<DataType> result = std::vector<std::remove_const<DataType>::type>(data.begin(), data.end());
+
+    if (order == 0 || length < 2 || order >= length)
```

Comment:
This could be optimized to
```C++
uint16_t skip = 0 - !(length < 2 || order >= length);
order = order & skip;
```
That way order will be equal to 0 whenever the condition fails, meaning the loop wouldn't execute. 
However, it'd be nice to explain in comment what does that operation do.